### PR TITLE
Feat/video preview

### PR DESCRIPTION
**Pull request by fredrikved**
A simple video preview, exposed in the admin gql endpoint, and implemented as a presentation interface in Directus.

Can implement a stream selector instead of defaulting to hls/dash.

### CODE REVIEW

File: cms/.template.env [hunk 1]
```diff
@@ -87,4 +87,9 @@ PUBSUB_PROJECT_ID=<REPLACE>
...
 
 API_PATH=http://localhost:8077/
-API_SECRET=123456
\ No newline at end of file
+API_SECRET=123456
+
+####################################################################################################
+## Content
+
+CONTENT_SECURITY_POLICY_DIRECTIVES__MEDIA_SRC=array:<TRUSTED_CDNS>
```
[the remark this answers is not in the document]
Comment (KillerX) [reply cms/.template.env:95]: Also is this used anywhere?

File: cms/src/endpoints/preview/index.ts [hunk 2]
```diff
@@ -61,6 +61,48 @@ const endpointConfig: EndpointConfig = {
...
+                        "X-Api-Key": apiSecret
+                    }
+                })
+
+                res.status(200).send(result.data.data.preview.asset)
+                return
+            } catch (e) {
+                console.log(e);
+                console.log("Couldn't fetch data from API")
+            }
```
Comment (KillerX) [inline cms/src/endpoints/preview/index.ts:104]: Maybe abstract this into a func that accepts the query and params?

**Review by KillerX — APPROVED** [review]
I think it looks ok on a high level. Not sure if there are any special things

File: cms/.template.env (hunk 1 above)
[the remark this answers is not in the document]
Comment (fredrikved) [reply cms/.template.env]: This env variable is used by Directus to set the media-src content security policy

**Review by KillerX — APPROVED** [review]

**Pull request merged**